docs/book/best-practice-identity.md

```diff
@@ -0,0 +1,123 @@
+* OPA let's you check the TTL.
+* OPA has early support for making HTTP requests during evaluation, which could be used to check if a JWT has been revoked.  Though if you're connecting to a remote system on every policy decision anyway, you should think about whether connecting to LDAP/AD directly is more appropriate (see below).
+
+
+
+## 2. Downloading LDAP/AD data as part of a policy bundle
+
+JWTs may not be available to you.  Or perhaps your policies require information about a user other than the one performing the action that OPA is authorizing.  For example, the policy you want to enforce says that either the owner of a resource or anyone in that owner's group may modify that resource.  So if Alice tries to modify a resource and she is not the owner, OPA would need to find the owner of the resource (say Bob) and look up all of his groups from LDAP/AD in order to check if Alice is in the same group as Bob.  OPA only has a JWT token for Alice, so it does not have group information for Bob.
+
+In any case, another option is to replicate LDAP/AD data in bulk into OPA.  One way to do that is through OPA's bundle feature, which periodically downloads policy bundles from a centralized server.  Those bundles can include data as well as policy.  If you implement the centralized server, you can include LDAP/AD data within the bundle.  Then every time OPA gets updated policies, it gets updated LDAP/AD data too.
+
+### Flow
+Two things happen independently with this kind of LDAP/AD integration.
+* (1) OPA downloads new policy bundles including LDAP/AD
+* (A) OPA-enabled software system asks OPA for policy decisions
+
+![Bundle flow](images/best-practice-identity-bundle.png)
+
+### Updates
+The lag between an LDAP/AD update and OPA having the update is the sum of the lag for an update between LDAP/AD and the central server and the lag for an update between the central server an OPA.  So if LDAP/AD updates every 5 minutes, and OPA pulls an update every 2 minutes, then the total maximum lag is 7 minutes.  Unlike the JWT case, it is feasible that a user could perform an action requiring authorization on the OPA-enabled service before OPA has the appropriate LDAP/AD policy, but you can account for that when writing policy and reject any request when there is insufficient data.
+
+### Size limitations
+Unlike the JWT case, where OPA only stores 1 user's LDAP/AD data at a time, when synchronizing LDAP/AD, OPA stores all users' LDAP/AD data at once in memory.  Obviously this can be a problem with large LDAP/AD stores.  Because the centralized server handles both policy and data it could provide just the LDAP/AD data necessary for the policies.
+
+<!-- **Security**
+* Don't expose OPA's API except through localhost
+* Assuming LDAP/AD is the only context needed, use OPA's authentication/authorization to disable all of OPA's APIs except those needed by the OPA-enabled service. -->
+
+## 3. Pushing LDAP/AD into OPA
```

Comment:
What's the main reason for someone to use this approach? Will the pull approach from section 2 not work because they need to optimize for update latency? If so, I would put more emphasis on that difference here.

Reply:
Main one is that they're not using the bundle API for whatever reason.  Also mentioned the update latency.  Added text to clarify.